homeassistant/components/knx/config_flow.py

```diff
@@ -128,8 +138,22 @@ async def async_step_connection_type(
             CONF_KNX_TUNNELING: CONF_KNX_TUNNELING.capitalize(),
             CONF_KNX_ROUTING: CONF_KNX_ROUTING.capitalize(),
         }
-        self._found_gateways = await scan_for_gateways()
-        if self._found_gateways:
+
+        if isinstance(self, OptionsFlow) and (knx_module := self.hass.data.get(DOMAIN)):
+            xknx = knx_module.xknx
+        else:
+            xknx = XKNX()
+        self._gatewayscanner = GatewayScanner(
+            xknx, stop_on_found=0, timeout_in_seconds=2
+        )
+        # keep a reference to the generator to avoid it being garbage collected before timeout
+        self._async_scan_gen = self._gatewayscanner.async_scan()
+        try:
+            assert self._async_scan_gen is not None
```

Comment:
This assert should be outside the try. But why is it even needed? Why can the function return None it should just return an empty iterator.

Reply:
Right, was not needed. Maybe a leftover 🙃